pageserver/src/page_service.rs

```diff
@@ -617,7 +651,7 @@ impl PageServerHandler {
                     });
 
                     pgb.write_message_noflush(&BeMessage::CopyData(&response_msg.serialize()))?;
-                    self.flush_cancellable(pgb, &timeline.cancel).await?;
+                    self.flush_cancellable(pgb, None).await?;
```

Comment:
Why not as on line 536?
```suggestion
                    self.flush_cancellable(pgb, &tenant.cancel).await?;
```

Reply:
Hmm, yes, you're right.  Usually in this path we would already have looked up a timeline (so `None` would still be safe, we'd be waiting on the timeline's cancellation token), but in the corner case that we are sending a PagestreamBeMessage that is the result of something like a missing timeline, we should respect the tenant's cancellation token.